### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by ShayDavidson**

### CODE REVIEW

File: PSix.iml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module external.linked.project.path="$MODULE_DIR$" external.root.project.path="$MODULE_DIR$" external.system.id="GRADLE" external.system.module.group="" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
+<module external.linked.project.path="$MODULE_DIR$" external.root.project.path="$USER_HOME$/Projects/PSix" external.system.id="GRADLE" external.system.module.group="" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
```
Comment (ShayDavidson) [inline PSix.iml]: revert this

File: app/app.iml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module external.linked.project.path="$MODULE_DIR$" external.root.project.path="$MODULE_DIR$/.." external.system.id="GRADLE" external.system.module.group="PSix" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
+<module external.linked.project.path="$MODULE_DIR$" external.root.project.path="$USER_HOME$/Projects/PSix" external.system.id="GRADLE" external.system.module.group="PSix" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
```
Comment (ShayDavidson) [inline app/app.iml]: same

File: app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import com.paypal.psix.services.UserSession;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class OnboardingFragment extends Fragment {
+
+    private static final String LOG_TAG = "MainFragment";
+    private static final List<String> PERMISSIONS = Arrays.asList("public_profile");
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java]: user_events

File: app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                updateUserData();
+            }
+        });
+
+        return view;
+    }
+
+    private void updateUserData() {
+        if (user != null) {
+            UserSession.setUser(user);
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java:74]: :+1:

File: app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        uiHelper.onPause();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        uiHelper.onDestroy();
+    }
+
+    private void onSessionStateChange(Session session, SessionState state, Exception e) {}
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java:98]: implementation mandatory?

File: app/src/main/java/com/paypal/psix/services/UserSession.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        if (fbId != null) {
+            return new User(
+                    fbId,
+                    getSharedPrefs().getString(CUR_USER_FIRST_NAME_KEY, null),
+                    getSharedPrefs().getString(CUR_USER_LAST_NAME_KEY, null)
+            );
+        }
+
+        return null;
+    }
```
Comment (ShayDavidson) [inline app/src/main/java/com/paypal/psix/services/UserSession.java:50]: what do you think off extending the api with these methods:
`isUserSignedIn`
`logout` (sets user to null).

File: build.gradle [hunk 7]
```diff
@@ -4,6 +4,9 @@ buildscript {
     repositories {
         jcenter()
     }
+    repositories {
+        mavenCentral()
+    }
```
Comment (ShayDavidson) [inline build.gradle]: is this needed?

**Comment by ShayDavidson** [comment]
just add the switch between the onboarding and events screen
and then remove the debug option from the menu (and the corresponding strings).

File: build.gradle (hunk 7 above)
Comment (avihut) [inline build.gradle]: Removed. I initially added it due to the instructions on integrating Facebook, but now I see it only needs to be in the app's gradle (which it already is), and not in the root gradle.

File: PSix.iml (hunk 1 above)
Comment (avihut) [inline PSix.iml]: fixed

File: app/app.iml (hunk 2 above)
Comment (avihut) [inline app/app.iml]: fixed

File: app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java (hunk 3 above)
Comment (avihut) [inline app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java]: Added

File: app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java (hunk 5 above)
Comment (avihut) [inline app/src/main/java/com/paypal/psix/fragments/OnboardingFragment.java:98]: Copied from the FacebookHello example, and it seems it no longer is.

Removed :)

File: app/src/main/java/com/paypal/psix/services/UserSession.java (hunk 6 above)
Comment (avihut) [inline app/src/main/java/com/paypal/psix/services/UserSession.java:50]: Added `isUserSignedIn`. It's perfectly legitimate that after the logout process is properly performed the ID will contain null so as to indicate no user is logged in.

**Pull request merged**